extension/persistence/eclipselink/src/main/java/org/apache/polaris/extension/persistence/impl/eclipselink/PolarisEclipseLinkMetaStoreSessionImpl.java

```diff
@@ -98,26 +99,42 @@ public PolarisEclipseLinkMetaStoreSessionImpl(
       @Nullable String confFile,
       @Nullable String persistenceUnitName,
       @Nonnull PrincipalSecretsGenerator secretsGenerator) {
+    this(
+        createEntityManagerFactory(realmContext, confFile, persistenceUnitName),
+        store,
+        storageIntegrationProvider,
+        secretsGenerator);
     LOGGER.debug(
         "Creating EclipseLink Meta Store Session for realm {}", realmContext.getRealmIdentifier());
-    emf = createEntityManagerFactory(realmContext, confFile, persistenceUnitName);
-
-    // init store
-    this.store = store;
     try (EntityManager session = emf.createEntityManager()) {
       this.store.initialize(session);
     }
+  }
+
+  private PolarisEclipseLinkMetaStoreSessionImpl(
+      EntityManagerFactory emf,
+      PolarisEclipseLinkStore store,
+      PolarisStorageIntegrationProvider storageIntegrationProvider,
+      PrincipalSecretsGenerator secretsGenerator) {
+    this.emf = emf;
+    this.store = store;
     this.storageIntegrationProvider = storageIntegrationProvider;
     this.secretsGenerator = secretsGenerator;
   }
 
+  @Override
+  public BasePersistence copyOf() {
+    return new PolarisEclipseLinkMetaStoreSessionImpl(
+        emf, store, storageIntegrationProvider, secretsGenerator);
```

Comment:
If we copy a new object with references to the original fields, why do we need need a copy? I am trying to understand the reason here.

Reply:
> [...] why do we need need a copy?

Because of this discussion: https://github.com/apache/polaris/pull/1166#discussion_r1992378463

I'm ok returning `this`, but people had concerns with method name in that case.

Reply:
What about names like `detach()`,  `snapshot()`?

Reply:
`detach()` works for me.

Reply:
I'm going to use `detach()` and avoid copies for existing implementations.

Reply:
reverted copies and renamed